@kiva/kv-components/utils/loanCard.js

```diff
@@ -78,65 +83,75 @@ export function loanCardComputedProperties(props) {
+		const themesData = loan.value?.themesData ?? [];
+
 		// P1 Category
 		// Exp limited to: Eco-friendly, Refugees and IDPs, Single Parents
 		// Tag as first option for LSE loans
 		if (isLseLoan && tags.length) {
 			const position = Math.floor(Math.random() * tags.length);
 			const p1Tag = tags[position];
-			callouts.push(p1Tag);
+			const tagData = findCalloutData(tagsData, p1Tag);
+			const id = tagData?.id ?? null;
+			callouts.push({ label: p1Tag, type: 'tag', id });
 		}
 
 		if (!categoryPageName.value) {
 			if (categories.ecoFriendly
 				// eslint-disable-next-line max-len
-				&& !callouts.find((c) => c.toUpperCase() === ECO_FRIENDLY_KEY || c.toUpperCase() === SUSTAINABLE_AG_KEY)) {
-				callouts.push('Eco-friendly');
+				&& !callouts.find((c) => c.label.toUpperCase() === ECO_FRIENDLY_KEY || c.label.toUpperCase() === SUSTAINABLE_AG_KEY)) {
+				callouts.push({ label: 'Eco-friendly' });
 			} else if (categories.refugeesIdps) {
-				callouts.push('Refugees and IDPs');
+				callouts.push({ label: 'Refugees and IDPs' });
 			} else if (categories.singleParents
-				&& !callouts.find((c) => c.toUpperCase() === SINGLE_PARENT_KEY)) {
-				callouts.push('Single Parent');
+				&& !callouts.find((c) => c.label.toUpperCase() === SINGLE_PARENT_KEY)) {
+				callouts.push({ label: 'Single Parent' });
 			}
 		}
 
 		// P2 Activity
-		if (activityName && categoryPageName.value?.toUpperCase() !== activityName.toUpperCase()) {
-			callouts.push(activityName);
+		if (activityName && activityId && categoryPageName.value?.toUpperCase() !== activityName.toUpperCase()) {
+			callouts.push({ id: activityId, label: activityName, type: 'activity' });
 		}
 
 		// P3 Sector
-		if (sectorName
+		if (sectorName && sectorId
 			&& (activityName.toUpperCase() !== sectorName.toUpperCase())
 			&& (sectorName.toUpperCase() !== categoryPageName.value?.toUpperCase())
 			&& callouts.length < 2) {
-			callouts.push(sectorName);
+			callouts.push({ id: sectorId, label: sectorName, type: 'sector' });
 		}
 
 		// P4 Tag
 		if (!!tags.length && callouts.length < 2) {
 			const position = Math.floor(Math.random() * tags.length);
 			const p4Tag = tags[position];
-			if (!callouts.filter((c) => c.toUpperCase() === p4Tag.toUpperCase()).length) {
-				callouts.push(p4Tag);
+			const tagData = findCalloutData(tagsData, p4Tag);
+			const id = tagData?.id ?? null;
+			if (!callouts.filter((c) => c.label.toUpperCase() === p4Tag.toUpperCase()).length) {
+				callouts.push({ label: p4Tag, type: 'tag', id });
 			}
 		}
 
 		// P5 Theme
 		if (!!themes.length && callouts.length < 2) {
 			const position = Math.floor(Math.random() * themes.length);
 			const theme = themes[position];
-			if (!callouts.filter((c) => c.toUpperCase() === theme.toUpperCase()).length
+			const themeData = findCalloutData(themesData, theme);
+			const id = themeData?.id ?? null;
+			if (!callouts.filter((c) => c.label.toUpperCase() === theme.toUpperCase()).length
```

Comment:
Same comment about checking if `id` is `null`

Reply:
We need null ids to verify on component side if clickable; this for custom and partners tags. Same above.